server-spi-private/src/main/java/org/keycloak/device/DeviceActivityManager.java

```diff
@@ -104,45 +99,16 @@ private static DeviceRepresentation getDeviceFromUserAgent(KeycloakSession sessi
         DeviceRepresentation current;
 
         try {
-            Client client = UA_PARSER.parse(userAgent);
-            current = new DeviceRepresentation();
-
-            current.setDevice(client.device.family);
-
-            String browserVersion = client.userAgent.major;
+            UserAgentAnalyzer uaa = UserAgentAnalyzer.newBuilder().hideMatcherLoadStats().withCache(10000).build();
```

Comment:
Note that this step takes a few seconds.
It is common to instantiate the UserAgentAnalyzer once and use it many times.